### PR TITLE
closes #365 Added checkstyle rule to restrict what Accumulo code is used...

### LINKED ISSUE #365: Analyze Fluo code to see what non-public Accumulo APIs are used
Remove any non-public Accumulo API dependencies if found and possible.  This was a todo found while working on #235.

### PR DESCRIPTION
**Pull request by keith-turner**
....  Stopped using some non-public Accumulo code

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -220,6 +225,34 @@
...
+              </module>
+            </module>
+          </checkstyleRules>
+          <failOnViolation>true</failOnViolation>
+         </configuration>
+         <executions>
+           <execution>
+             <id>check-style</id>
+             <goals>
+               <goal>check</goal>
```
Comment (mikewalch) [inline pom.xml]: I imported a non-public class and received a failure when I ran `mvn checkstyle:check`.  However, I didn't get a failure when I ran `mvn verify`.  What do you think about adding this check to verify?

Comment (keith-turner) [inline pom.xml]: I was seeing failures when running `mvn verify`

**Comment by keith-turner** [comment]
I just made the following change and ran `mvn clean verify`

```
diff --git a/modules/accumulo/src/main/java/io/fluo/accumulo/data/MutableBytesFactory.java b/modules/accumulo/src/main/java/io/fluo/accumulo/data/MutableBytesFactory.java
index ee793e7..450a999 100644
--- a/modules/accumulo/src/main/java/io/fluo/accumulo/data/MutableBytesFactory.java
+++ b/modules/accumulo/src/main/java/io/fluo/accumulo/data/MutableBytesFactory.java
@@ -16,11 +16,13 @@
 package io.fluo.accumulo.data;
 
 import io.fluo.api.data.Bytes;
+import org.apache.accumulo.core.gc.thrift.GCStatus;
 
 /**
  * An implementation of BytesFactory
  */
 public class MutableBytesFactory implements Bytes.BytesFactory {
+  private GCStatus gcs;
   
   @Override
   public Bytes get(byte[] data) {

```

The build failed w/ the following 

```
[INFO] There are 1 checkstyle errors.
[ERROR] src/main/java/io/fluo/accumulo/data/MutableBytesFactory.java[19] (regexp) RegexpSinglelineJava: Accumulo non-public classes imported
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Fluo Parent ....................................... SUCCESS [4.570s]
[INFO] Fluo API .......................................... SUCCESS [18.743s]
[INFO] Fluo Accumulo Extensions .......................... FAILURE [13.230s]
[INFO] Fluo Core ......................................... SKIPPED
[INFO] Fluo Metrics ...................................... SKIPPED
[INFO] Fluo MapReduce .................................... SKIPPED
[INFO] Fluo Cluster ...................................... SKIPPED
[INFO] Fluo Distribution ................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

```

File: pom.xml (hunk 1 above)
Comment (mikewalch) [inline pom.xml]: Yes.  You are right.  It does fail.  I am not sure why I didn't see it the first time.

**Comment by mikewalch** [comment]
+1

**Pull request merged**